malwarefront/config-overrides.js

```diff
@@ -1,7 +1,76 @@
-const { override, babelInclude, removeModuleScopePlugin } = require('customize-cra');
+const { override, babelInclude, removeModuleScopePlugin, addWebpackPlugin, addWebpackModuleRule, addWebpackResolve } = require('customize-cra');
+const fs = require("fs");
 const path = require('path');
+const webpack = require('webpack');
 
-module.exports = override(
-  removeModuleScopePlugin(),
-  babelInclude([path.resolve('src'), path.resolve(__dirname, '../plugins')])
-);
\ No newline at end of file
+const pluginPackagePrefix = "@malwarefront-plugin";
+const pluginsIndexFile = path.resolve(require.resolve("@malwarefront/extensions"), '..', 'plugins.js');
+
+function findInstalledPlugins() {
+    let modules = {};
+    let modulesDir = path.resolve(__dirname, "node_modules", pluginPackagePrefix);
+    if(!fs.existsSync(modulesDir))
+        return {}
+    for(let pluginName of fs.readdirSync(modulesDir)) {
+        let realPath = path.resolve(modulesDir, pluginName);
+        try {
+            realPath = path.resolve(modulesDir, fs.readlinkSync(realPath));
+        } catch(e) {
+            if(e.errno != -22 /* EINVAL */)
+                throw e;
+        }
+        modules[pluginPackagePrefix + "/" + pluginName] = realPath;
+    }
+    return modules;
+}
+
+function getPluginPaths(plugins) {
+    return Object.values(plugins)
+}
+
+function getPluginLoaders(plugins) {
+    let moduleNames = Object.keys(plugins)
+    return moduleNames.map((moduleName) => ({
+        loader: 'imports-loader',
+        options: {
+            imports: [
+                {
+                    moduleName,
+                    name: moduleName.split("/")[1]
+                }
+            ]
+        }
+    })).concat({
+        loader: 'exports-loader',
+        options: {
+            exports: moduleNames.map((moduleName) => ({
+                syntax: 'named',
+                name: moduleName.split("/")[1]
+            }))
+        }
+    })
+}
+
+const plugins = findInstalledPlugins();
+
+for(let pluginName of Object.keys(plugins)) {
+    console.log("[plugins] Found plugin "+pluginName);
+}
+
+module.exports = {
+    webpack: override(
+        removeModuleScopePlugin(),
+        babelInclude([path.resolve('src')].concat(getPluginPaths(plugins))),
+        (
+            /* If there are no plugins: don't add any loaders */
+            Object.keys(plugins).length === 0 ?
```

Comment:
The JS-way would be to do `Object.keys(plugins).length` and flip operands, but I guess this is more verbose.

Reply:
Agree, fixed.